cpp-ch/local-engine/Storages/SubstraitSource/TextFormatFile.cpp

```diff
@@ -1,107 +1,36 @@
+#include <Processors/Formats/Impl/HiveTextRowInputFormat.h>
 
 namespace local_engine
 {
-namespace ErrorCodes
-{
-    extern const int NOT_IMPLEMENTED;
-}
 
-static DB::FormatSettings updateFormatSettings(const DB::FormatSettings & settings, const DB::Block & header)
+TextFormatFile::TextFormatFile(
+    DB::ContextPtr context_, const substrait::ReadRel::LocalFiles::FileOrFiles & file_info_, ReadBufferBuilderPtr read_buffer_builder_)
+    : FormatFile(context_, file_info_, read_buffer_builder_)
 {
-    DB::FormatSettings updated = settings;
-    updated.skip_unknown_fields = true;
-    updated.with_names_use_header = true;
-    updated.date_time_input_format = DB::FormatSettings::DateTimeInputFormat::BestEffort;
-    updated.csv.delimiter = updated.hive_text.fields_delimiter;
-    if (settings.hive_text.input_field_names.empty())
-        updated.hive_text.input_field_names = header.getNames();
-    return updated;
 }
 
-TextFormatFile::TextFormatFile(DB::ContextPtr context_, const substrait::ReadRel::LocalFiles::FileOrFiles & file_info_, ReadBufferBuilderPtr read_buffer_builder_)
-    :FormatFile(context_, file_info_, read_buffer_builder_) {}
-
 FormatFile::InputFormatPtr TextFormatFile::createInputFormat(const DB::Block & header)
 {
     auto res = std::make_shared<FormatFile::InputFormat>();
     res->read_buffer = std::move(read_buffer_builder->build(file_info, true));
-    DB::FormatSettings format_settings = DB::getFormatSettings(context);
-    format_settings.with_names_use_header = true;
-    format_settings.skip_unknown_fields = true;
-    std::string text_field_delimiter = file_info.text().field_delimiter();
-    size_t max_block_size = file_info.text().max_block_size();
-    format_settings.hive_text.fields_delimiter = *text_field_delimiter.data();
-    DB::RowInputFormatParams in_params = {max_block_size};
-    std::shared_ptr<local_engine::TextRowInputFormat> txt_input_format = 
-        std::make_shared<local_engine::TextRowInputFormat>(header, *(res->read_buffer), in_params, format_settings, file_info.text().schema());
-    res->input = txt_input_format;
-    return res;
-}
-
-TextRowInputFormat::TextRowInputFormat(
-    const DB::Block & header_, 
-    DB::ReadBuffer & in_, 
-    const DB::RowInputFormatParams & params_, 
-    const DB::FormatSettings & format_settings_,
-    const substrait::NamedStruct & input_schema_)
-    : TextRowInputFormat(header_, std::make_unique<DB::PeekableReadBuffer>(in_), params_, updateFormatSettings(format_settings_, header_))
-{
-    input_schema = input_schema_;
-}
 
-TextRowInputFormat::TextRowInputFormat(
-    const DB::Block & header_, std::shared_ptr<DB::PeekableReadBuffer> buf_, const DB::RowInputFormatParams & params_, const DB::FormatSettings & format_settings_)
-    : DB::CSVRowInputFormat(
-        header_, buf_, params_, true, false, format_settings_, std::make_unique<TextFormatReader>(*buf_, format_settings_))
-{
-}
-
-void TextRowInputFormat::readPrefix()
-{
```

Comment:
This readPrefix() can not be removed, it was introduced to slove the problem the field index error,  And you can test the sql `select count(*) from $table where id < ''` can reproduce this

Reply:
column mapping is implemented through hive_text.input_field_names.